core/templates/pages/story-editor-page/editor-tab/story-editor.directive.ts

```diff
@@ -231,6 +234,14 @@ angular.module('oppia').directive('storyEditor', [
             StoryUpdateService.setStoryTitle($scope.story, newTitle);
           };
 
+          $scope.updateStoryUrlFragment = function(newUrlFragment) {
+            if (newUrlFragment === $scope.story.getUrlFragment()) {
+              return;
+            }
+            StoryUpdateService.setStoryUrlFragment(
```

Comment:
Do you need to check uniqueness at this point?

Reply:
We would need to do a backend call for that everytime this method is called (on blur event), I felt that would be too much computation (because we check uniqueness across all stories on the server).. There is a backend validation for this in place so it would fail with an appropriate error message on trying to save changes. Do you still think we should have one check in the editor UI?

Reply:
Blur events on this field won't happen that often, so I think it's fine to add a backend call. 

Also, as a general note, it is always better to prevent errors from getting to the backend in the first place.

Reply:
Done.